picocrt/machine/arm/crt0.c

```diff
@@ -132,6 +197,55 @@ _start(void)
 	__asm__("vmsr fpexc, %0" : : "r" (0x40000000));
 #endif
 
+#ifdef _PICOCRT_ENABLE_MMU
+
+#if __ARM_ARCH >= 7 && __ARM_ARCH_PROFILE != 'R'
+
+#define SCTLR_MMU (1 << 0)
+#define SCTLR_DATA_L2 (1 << 2)
+#define SCTLR_BRANCH_PRED (1 << 11)
+#define SCTLR_ICACHE (1 << 12)
+
+        uint32_t        mmfr0;
```

Comment:
I am getting `../../picolibc/picocrt/machine/arm/crt0.c:209:9: error: non-ASM statement in naked function is not supported` with Clang 16.

Reply:
oh, right. I also did a tiny computation in C code to get exceptions running in Thumb on some targets. I wonder if it would be easier to just move all of this into _cstart?

Reply:
I don't see any harm in doing it from C code.

Reply:
Yeah, that was a bunch easier and leave all of the control flow in C, which I like. Let me know if this works for you as well; I'll see about testing it with clang here.